### PR TITLE
Docs site cleanup

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Convert docs site to be component based!
- Big cleanup in anticipation of upstreaming back into `example/docs`
- Based on #945, but no code/behavior changed. Just refactoring & cleanup.

This is one of those PRs that's going to create constant merge conflicts, given that it touches everything. So I'll most likely push to get it merged sooner, hopefully by end of day tomorrow. If the bug mentioned in #945 hasn't been fixed by tomorrow afternoon, then I'll update this PR and comment out the language dropdown component so that this PR will have no impact on the user.

### CODE REVIEW

File: docs/src/components/RightSidebar/RightSidebar.astro [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    top: 0;
+  }
+  .sidebar-nav-inner {
+    height: 100%;
+    padding: 0;
+    padding-top: var(--doc-padding);
+    overflow: auto;
+  }
+</style>
+<nav class="sidebar-nav" aria-labelledby="sidebar-content">
```
Comment (jasikpark) [inline docs/src/components/RightSidebar/RightSidebar.astro:20]: this aria-labelledby needs to be updated to the id of the element that's wrapping it in MainLayout.astro

**Comment by jasikpark** [comment]
Left some comments.. It looks pretty great though!

**Review by jasikpark — CHANGES_REQUESTED** [review]

File: docs/src/components/RightSidebar/RightSidebar.astro (hunk 1 above)
Comment (FredKSchott) [reply docs/src/components/RightSidebar/RightSidebar.astro]: great catch

File: docs/public/index.css [hunk 2]
```diff
@@ -183,13 +183,14 @@ strong {
 }
 
 /* Supporting Content */
+code {
+  font-family: var(--font-mono);
+}
 
 code:not([class*='language']) {
```
Comment (jasikpark) [inline docs/public/index.css:187]: Would it make sense to resolve https://github.com/snowpackjs/astro/issues/760 here? Or maybe that should be a different PR especially since Nate hasn't commented on whether that was intentional...

File: docs/src/components/Header/SkipToContent.astro [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  display: block;
+  text-align: center;
+  background-color: var(--theme-text-accent);
+  color: var(--theme-bg);
+  border-radius: 0.25em;
+  outline: var(--theme-bg) solid 1px;
+  outline-offset: 0;
+}
+</style>
+<a href="#article" class="sr-only focus:not-sr-only skiplink"><span>Skip to Content</span></a>
```
Comment (jasikpark) [inline docs/src/components/Header/SkipToContent.astro:21]: i'm not sure whether "focus:not-sr-only" exists in the project 🤔 i don't remember whether I ended up adding it..

File: docs/src/components/RightSidebar/ThemeToggleButton.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import type { FunctionalComponent } from 'preact';
 import { h, Fragment } from 'preact';
 import { useState, useEffect } from 'preact/hooks';
+import './ThemeToggleButton.css';
 
 const themes = ['light', 'dark'];
```
Comment (jasikpark) [inline docs/src/components/RightSidebar/ThemeToggleButton.tsx:6]: I'm not sure this is a comment for this PR, but I'd really appreciate it if we had a way to unset your theme preference and just use "prefers-color-scheme" for the theme switcher, since I often go to sites and fiddle with the theme to check out the colors, but then I just want it to follow my system after that!

File: docs/src/config.ts [hunk 5]
```diff
@@ -50,36 +50,20 @@ export const sidebar = [
...
-  //   ],
-  // },
 ];
 
-export const site = {
+export const SITE = {
   title: 'Astro Documentation',
   description: 'Build faster websites with less client-side Javascript.',
-  ogLocale: 'en_US',
+}
```
Comment (jasikpark) [inline docs/src/config.ts:58]: to provide context for why this was named ogLocale and not "locale" is because it's a different locale formatting than what you would use in the lang="" attribute on html: "en-us" vs "en_US" - I think your refactoring works great though!

File: docs/src/config.ts [hunk 6]
```diff
@@ -50,36 +50,20 @@ export const sidebar = [
...
+  locale: 'en_US',
   image: {
     src: '/default-og-image.png?v=1',
     alt:
       'astro logo on a starry expanse of space,' +
       ' with a purple saturn-like planet floating in the right foreground',
   },
-  twitter: {
-    site: 'astrodotbuild',
-    creator: 'astrodotbuild',
```
Comment (jasikpark) [inline docs/src/config.ts:83]: It'd be really cool if we could generate this list of contributor's twitter handles for a docs page, since "twitter:creator" is meant to signify the handle of the author of the linked content 😆. I doubt Twitter even has multi-author support

File: docs/src/layouts/SplashLayout.astro [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        flex-direction: column;
+        height: 100%;
+      }
+    </style>
+  </head>
+
+  <body>
+    <Header />
+    <main class="layout splash-layout">
+      <article>
```
Comment (jasikpark) [inline docs/src/layouts/SplashLayout.astro:43]: I assume it's intentional to have no sidebar nav here?

I think that's a valid stylistic choice probably though i'd double check that we aren't shipping a broken "SidebarToggleButton" in "Header" that doesn't have a sidebar nav to toggle 😆

File: docs/src/components/Header/Header.astro [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  .search-item {
+    max-width: 400px;
+  }
+}
+</style>
+<header>
+    <SkipToContent />
+    <nav class="nav-wrapper" title="Top Navigation">
+    <div class="menu-toggle">
+        <SidebarToggle client:idle/>
```
Comment (jasikpark) [inline docs/src/components/Header/Header.astro:125]: The sidebar toggle doesn't have active or hover styles anymore, maybe could the global button design be changed?

**Review by jasikpark — CHANGES_REQUESTED** [review]
I'm really excited to see this get merged! There's some comments that don't necessarily need to be resolved in this PR, but I left them anyways 😅

File: docs/src/components/Header/SkipToContent.astro (hunk 3 above)
Comment (FredKSchott) [reply docs/src/components/Header/SkipToContent.astro:21]: nope, I don't see it. removed!

File: docs/src/config.ts (hunk 5 above)
Comment (FredKSchott) [reply docs/src/config.ts:58]: Yea good call. the intention is that it's `OPEN_GRAPH.locale` which hopefully makes that clear as well.

File: docs/public/index.css (hunk 2 above)
Comment (FredKSchott) [reply docs/public/index.css:187]: Makes sense to me, I'll do it without tackling a larger CSS `code / pre` refactor just yet
